mailbox/backup/src/test/java/org/apache/james/mailbox/backup/ZipperTest.java

```diff
@@ -196,6 +201,26 @@ void archiveShouldWriteMailboxAnnotationsMetadataWhenTwoPresent() throws Excepti
         }
     }
 
+    private MailboxAnnotation getAnnotationForIndex(int i) {
+      return MailboxAnnotation.newInstance(new MailboxAnnotationKey("/annotation" + i + "/test"), "annotation " + i + " content");
+    }
+
+    private final List<MailboxAnnotation> oneThousandAnnotations =   ContiguousSet.create(Range.closed(0, 999), DiscreteDomain.integers()).stream()
+            .map(this::getAnnotationForIndex).collect(Collectors.toList());
+
+    @Test
+    void archiveShouldWriteMailboxAnnotationsMetadataWhenOneThousandPresent() throws Exception {
+        testee.archive(ImmutableMap.of(MAILBOX_1, oneThousandAnnotations), Stream.of(), output);
+
+        try (ZipFile zipFile = new ZipFile(toSeekableByteChannel(output))) {
+            assertThatZip(zipFile)
+                .containsOnlyEntriesMatching(
+                    hasName(MAILBOX_1.getName() + "/")
+                        .containsExtraFields(
+                            new MailBoxAnnotationsExtraField(oneThousandAnnotations)));
+        }
+    }
+
```

Comment:
can you add a test with a too long annotation?

Reply:
Since it's in the same extra field, it think that the test with one thousand annotations serves this very purpose. No ?

Reply:
yes, but your test is not failing. I would expect a failing test (maybe use 10 000 or 100 000 annotations if you prefer). If I remember well extra fields are limited to 64k.